backend/bitsurance/bitsurance.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bitsurance
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/accounts"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/accounts/types"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/signing"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
+)
+
+const (
+	apiKey    = "265bfd773038c9ad9da7047e107babba0269bc3d31952172d9b10335e8a9d8e9"
+	xpubSalt  = "bitsurance"
+	apiURL    = "https://api.bitsurance.eu/api/"
+	widgetURL = "https://api.bitsurance.eu/widget/"
+)
+
+// fetchBitsuranceIds fetches the list of insured account hashes from the Bitsurance
+// server. The result is returned as a slice of strings, with each element containing
+// the salted hash of an account xpub, in lexical order. (0-9a-z).
+// For details on the hashing please refer to `BitsuranceGetId`.
+func fetchBitsuranceIds(httpClient *http.Client) ([]string, error) {
+	endpoint := apiURL + "allAccountHashes"
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	req.Header.Add("X-API-KEY", apiKey)
+
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return nil, errp.WithStack(err)
+	}
+	defer res.Body.Close() //nolint:errcheck
+	if res.StatusCode != http.StatusOK {
+		return nil, errp.Newf("%s - bad response code %d", endpoint, res.StatusCode)
+	}
+	const max = 81920
```

Comment:
isn't this way to small of a limit?

Reply:
Probably yes, but we'll probably change this part calling a per-account endpoint instead of downloading the full list. I didn't change it as the whole function will probably be removed of heavily changed in the next PR